ServerSide/yoty/WebApi/Controllers/BidsController.cs

```diff
@@ -23,57 +23,160 @@ public BidsController(IBidsManager bidsManager)
+        }
+
+
+        
+        [HttpPost]
         [Route("{bidId}/Buyers")]
-        public async Task<IList<BuyerDTO>> GetProductBidBuyers(string bidId)
+        public async Task<ActionResult<BuyerDTO>> AddBuyer(BidBuyerJoinRequest bidBuyerJoinRequest)
         {
-            return await bidsManager.GetBidBuyers(bidId);
+
+            Response<BuyerDTO> response = await this.bidsManager.AddBuyer(bidBuyerJoinRequest).ConfigureAwait(false);
+            if (response.IsOperationSuccsseded)
+            {
+                return this.StatusCode(StatusCodes.Status201Created, response.DTOObject);
+            }
+            return this.StatusCode(StatusCodes.Status304NotModified, response.SuccessOrFailureMessage);
         }
-        /*
+
         [HttpPost]
+        [Route("{bidId}/Proposals")]
+        public async Task<ActionResult<SupplierProposalDTO>> AddSupplierProposal(SupplierProposalRequest supplierProposalRequest)
+        {
+
+            Response<SupplierProposalDTO> response = await this.bidsManager.AddSupplierProposal(supplierProposalRequest).ConfigureAwait(false);
+            if (response.IsOperationSuccsseded)
+            {
+                return this.StatusCode(StatusCodes.Status201Created, response.DTOObject);
+            }
+            return this.StatusCode(StatusCodes.Status304NotModified, response.SuccessOrFailureMessage);
+        }
+
+        [HttpPut]
+        [Route("{bidId}")]
+        public async Task<ActionResult<BidDTO>> EditBid(EditBidRequest editBidRequest)
+        {
+
+            Response<BidDTO> response = await this.bidsManager.EditBid(editBidRequest).ConfigureAwait(false);
+            if (response.IsOperationSuccsseded)
+            {
+                return response.DTOObject;
+            }
+            return this.StatusCode(StatusCodes.Status304NotModified, response.SuccessOrFailureMessage);
+        }
+
+
+        [HttpDelete]
         [Route("{bidId}/Buyers/{buyerId}")]
-        public async Task<ActionResult<string>> PostNewBuyerToBid(string productBidId, string buyerId, BidBuyerJoinRequest bidBuyerJoinRequest)
+        public async Task<ActionResult> DeleteBuyer(string bidId, string buyerId)
         {
-            try
+
+            Response response = await this.bidsManager.DeleteBuyer(bidId, buyerId).ConfigureAwait(false);
+            if (response.IsOperationSuccsseded)
             {
-                await this.bidsManager.AddBuyerToBid(productBidId, buyerId, bidBuyerJoinRequest).ConfigureAwait(false);
-                return Ok(8); // checking
+                return this.StatusCode(StatusCodes.Status200OK, response.SuccessOrFailureMessage);
+
             }
 
-            catch (Exception e)
+            return this.StatusCode(StatusCodes.Status405MethodNotAllowed, response.SuccessOrFailureMessage);
+        }
+
+
+        [HttpDelete]
+        [Route("{bidId}/Proposals/{proposalId}")]
+        public async Task<ActionResult> DeleteSupplierProposal(string bidId, string proposalId)
```

Comment:
please use bidId and supplierId (instead of proposalId) here and in DeleteSupplierProposal of bids manager